src/cmd/pack.ts

```diff
@@ -74,30 +75,54 @@ export default async function pack(
       subManifest
     );
 
-    await stream
+    const carStream = stream
       .pipeThrough(
         new TransformStream<Block, Block>({
           transform(block, controller) {
             // we visit every block, the last one will be the root block
             rootCID = CID.parse(block.cid.toString());
 
-            // TODO: work out the proper piece CID (CommP)
-            pieceCID = rootCID;
             // console.log("Root CID:", rootCID.toString());
             controller.enqueue(block);
           },
         })
       )
-      .pipeThrough(new CAREncoderStream([placeholderCID]))
-      .pipeTo(
+      .pipeThrough(new CAREncoderStream([placeholderCID]));
+
+    // Annoyingly we now have to stream the whole lot again as Piece Commitment
+    // requires FR32 padding the finished CAR file, so fork it and process in parallel.
+    const [toFile, toCommP] = carStream.tee();
```

Comment:
One has to be a bit careful with tee() as it goes at the speed of the fastest consumer and the slower one caches in memory.  Can you check the memory usage with huge files with and without this?  (writing to disk tends to be a lot slower than "in memory" processing.  I considered using tee() for hashing the files while streaming them to the unixfs writer, but I found it better to call hash.upate() in the unixfs writer stream.  I don't know if that pattern would be appropriate here? (it works well when one consumer stream is a lot faster than the other)

Reply:
Will check. Useful context for a test, possibly. 

The alternative would be to let the writer write and then load the CAR back off disk but that felt less elegant.

Reply:
As agreed, we will get this change in and then review streaming performance